lib/jekyll/regenerator.rb

```diff
@@ -26,6 +26,7 @@ def regenerate?(document)
       when Document
         regenerate_document?(document)
       else
+        return false if document.is_a?(String) && file_in_theme_dir?(document)
```

Comment:
Where do we call `regenerate?` with a `String`?

Reply:
If you debug *right here* using `puts document.inspect` calls, you'll see that some of the dependencies (on the first build with `-I` flag) are just regular paths to files in the theme-gem